### ISSUE TITLE
RunSignalHandler: invalid signal handler

### ISSUE THREAD

**Issue by ttungl** [Multi2Sim/m2s-bench-heteromark#3]
Hi Yifan,
    When running the `aes` workload, I got the error as below. Do you know what kind of error and how to fix it? Thanks a lot.
Best,
```
; Multi2Sim 5.0 - A Simulation Framework for CPU-GPU Heterogeneous Computing
; Please use command 'm2s --help' for a list of command-line options.
; Simulation alpha-numeric ID: lTHBJ

terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument HeteroArchGen4M2S/benchmarks/m2s-bench-heteromark/aes/data

RunSignalHandler: invalid signal handler

tungle@tungle-Precision-T7600:~/Documents/multi2sim$ 
```

**Comment by syifan** [comment]
AES benchmark requires an input of the plaintext and an input of the key. Example input files are in the data folder alongside the executables. So you can run `./aes -h` to check what is the command line options for that two input. 

Also the executable compiled for Multi2Sim cannot run in native environment.

**Comment by ttungl** [comment]
Yes, from the benchmark.ini, I can set something like this in the benchmark part:
`./benchmarks/m2s-bench-heteromark/aes/aes_hsa -k benchmarks/m2s-bench-heteromark/aes/data/key.data -i benchmarks/m2s-bench-heteromark/aes/data/small.data benchmarks/m2s-bench-heteromark/aes/data`
And, I observe that there is no interaction with the GPU (southern islands) in `histogram` workload. Do I need to install something to enable it? I've learned that HIP could be used to enable HSA in multi2sim, is it correct? I followed this [link](https://github.com/RadeonOpenCompute/rocBLAS/wiki/Setting-up-enviroment-on-NVIDIA-platform). Am I in the right direction?

**Comment by ttungl** [comment]
I have a question that, can I run the compiled benchmark without installing the [prerequisite environment like HSA](https://github.com/NUCAR-DEV/Hetero-Mark/tree/v2.0-alpha-1) ? (my desktop uses NVIDIA quadro 4000 graphic card). I tried once, and it's crashed.

**Comment by syifan** [comment]
No, it is not possible. Even if you installed the HSA stack, you cannot run the benchmark there. The special compiled benchmark statically linked with the Multi2Sim HSA runtime and therefore will only connect to Multi2Sim HSA driver, rather than the system driver.

**Comment by ttungl** [comment]
One thing I still not figure out is that, I can run the simulation and get the result with `histogram` workload (`histo_hsa`), but the GPU part seems like no data transaction on it. I wonder that this is either the `histogram` workload does not have any sort of interactions with GPU cores, or because the simulator can't enable the Graphic card to do so. Please correct me if wrong. Thanks.

**Comment by syifan** [comment]
The Simulator is purely on CPU, it does not involve any interaction with your GPU card.

**Comment by ttungl** [comment]
I got it. Appreciate that Yifan.

**Comment by ttungl** [comment]
Hi Yifan, I got this error when running the network only mode:
`m2s --net-config ./HeteroArchGen4M2S/configs/netconfig --net-sim net-l2-mm --net-max-cycles 100000 --net-report aes_net_report.out HeteroArchGen4M2S/benchmarks/m2s-bench-heteromark/aes/aes_hsa -k HeteroArchGen4M2S/benchmarks/m2s-bench-heteromark/aes/data/key.data -i HeteroArchGen4M2S/benchmarks/m2s-bench-heteromark/aes/data/small.data HeteroArchGen4M2S/benchmarks/m2s-bench-heteromark/aes/data`

Error message:
```
terminate called after throwing an instance of 'std::invalid_argument'
  what():  Invalid argument HeteroArchGen4M2S/benchmarks/m2s-bench-heteromark/aes/data

RunSignalHandler: invalid signal handler
```
I would like to ask that what is the problem to this error? Do you know how to fix this? 
Thanks,
Tung

**Comment by syifan** [comment]
Sorry for the late reply. I think you put one extra argument at the end the simulation command. Can you try `m2s --net-config ./HeteroArchGen4M2S/configs/netconfig --net-sim net-l2-mm --net-max-cycles 100000 --net-report aes_net_report.out HeteroArchGen4M2S/benchmarks/m2s-bench-heteromark/aes/aes_hsa -k HeteroArchGen4M2S/benchmarks/m2s-bench-heteromark/aes/data/key.data -i HeteroArchGen4M2S/benchmarks/m2s-bench-heteromark/aes/data/small.data` instead?

**Comment by ttungl** [comment]
I thought the last one `data` path should be included into the command as guided in the `benchmark.ini` file.
I ran again with this command line, and got the error at the output as below.

#Command:
`m2s --net-config ./HeteroArchGen4M2S/configs/netconfig --net-sim net-l2-mm --net-max-cycles 100000 --net-report aes_net_report.out HeteroArchGen4M2S/benchmarks/m2s-bench-heteromark/aes/aes_hsa_yifan -k HeteroArchGen4M2S/benchmarks/m2s-bench-heteromark/aes/data/key.data -i HeteroArchGen4M2S/benchmarks/m2s-bench-heteromark/aes/data/small.data`

#Output:
```
[ELF Reader]
[/home/tungle/Documents/multi2sim/HeteroArchGen4M2S/benchmarks/m2s-bench-heterom
ark/aes/aes_hsa_yifan] Error: 64-bit ELF files not supported

The ELF file being loaded is a 64-bit file, currently not supported by
Multi2Sim. If you are compiling your own source code on a 64-bit machine, please
use the '-m32' flag in the gcc command-line. If you get compilation errors
related with missing '.h' files, check that the 32-bit gcc package associated
with your Linux distribution is installed.
```
